lib/components/T3CeFormFormframework/T3CeFormFormframework.vue

```diff
@@ -40,9 +40,29 @@ export default {
         serverSuccess: 'The form was sent, thank you.',
         serverError: 'We can not process form right now, please try again later.',
         validationErrors: 'There were some errors, review the form'
+      },
+      // map TYPO3 validation rules to
+      // https://vee-validate.logaretm.com/v3/guide/rules.html#rules
+      rules: {
+        Alphanumeric: 'alpha',
+        Integer: 'integer',
+        NotEmpty: 'required',
+        EmailAddress: 'email',
+        RegularExpression: {
+          identifier: 'regex',
+          options: {
+            expression: 'regex'
+          }
+        }
       }
     }
   },
+  computed: {
+    elements () {
+      const elements = [...this.form.elements]
```

Comment:
`this.form.elements` is an array of objects, so it will be passed by reference anyway. You can simplify it and instantly return `this.prepareElements(this.form.elements)`.

Reply:
right :)